### PR TITLE
[Merged by Bors] - docs: Add to instructions for building docker image

### PR DESCRIPTION
**Pull request by mfdorst**
I ran into a compilation error because I was missing the `x86_64-unknown-linux-musl` target. This adds instructions for installing that.

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -176,6 +176,9 @@ $ ./k8-util/cluster/reset-kind.sh
 #### Build Fluvio CLI and docker image from source code
 
 ```
+# Install the cross-compilation toolchain needed to compile the docker image (only required the first time)
+$ rustup target add x86_64-unknown-linux-musl
```
[the remark this answers is not in the document]
Comment (mfdorst) [reply DEVELOPER.md:180]: Oh good point

[the remark this answers is not in the document]
Comment (mfdorst) [reply DEVELOPER.md:181]: Updated

[the remark this answers is not in the document]
Comment (mfdorst) [reply DEVELOPER.md:181]: Wait that's not quite right either, hang on

[the remark this answers is not in the document]
Comment (mfdorst) [reply DEVELOPER.md:181]: Okay that's better

File: DEVELOPER.md [hunk 2]
```diff
@@ -176,6 +176,12 @@ $ ./k8-util/cluster/reset-kind.sh
 #### Build Fluvio CLI and docker image from source code
 
 ```
+# Install the cross-compilation toolchain needed to compile the docker image (only required the first time)
+# x86_64 (most computers):
+$ rustup target add x86_64-unknown-linux-musl
+# M1 Macs:
+$ rustup target add aarch-unknown-linux-musl
```
Comment (morenol) [inline DEVELOPER.md:183]: I think that it is `aarch64-unknown-linux-musl`

**Review by morenol — CHANGES_REQUESTED** [review]
Could you update the PR title to follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/)?

Something like:

`docs:  Add to instructions for building docker image` should work

**Comment by mfdorst** [comment]
Haha I did that last time and was told to do (docs) instead but sure.

File: DEVELOPER.md (hunk 2 above)
Comment (mfdorst) [reply DEVELOPER.md:183]: Oh shoot my bad. I'll fix it when I get home, thanks.

Comment (mfdorst) [reply DEVELOPER.md]: Fixed

**Review by morenol — APPROVED** [review]

**Review by tjtelan — APPROVED** [review]
LGTM

**Pull request closed without merging**